### ISSUE TITLE
Add EntityFramework support to Linq provider

### ISSUE THREAD

**Issue by jrote1** [bchavez/RethinkDb.Driver#159]
**Context**

It would be great if it was possible to use this driver with EntityFramework it would allow existing EntityFramework users to get into using RethinkDb

**Alternatives**

No

**Has the feature been requested before?**

No

**If the feature request is approved, would you be willing to submit a PR?**

Yes, I can add this feature once the Linq provider supports all the use cases I can think of

**Comment by bchavez** [comment]
Entity Framework support for RethinkDB sounds good!

I think we'd probably want to create a new assembly / nuget release like `RethinkDb.Driver.EntityFrameworkCore` (or something similar) that ultimately contains hard references to EF with glue types contained in the `RethinkDb.Driver.EntityFrameworkCore` assembly.

This way, the core `RethinkDb.Driver` driver and `RethinkDb.Driver.Linq` package don't contain hard references to EF core. Let me know what you think.

**Comment by jrote1** [comment]
Yeah that was my thinking as well to have a separate assembly for the EnityFramework related code

**Comment by jrote1** [comment]
@bchavez When are next online, need to ask you a few things?

**Comment by bchavez** [comment]
Hey Jake, I should be online for a few hours. 5 PM to about 7 or 8 PM Pacific Time / Los Angeles.

**Comment by ThaDaVos** [comment]
Any progress on this?
